core/search/fields.py

```diff
@@ -213,10 +213,14 @@ def get_condition(self, expression: Expression, remainder: List[str]) -> Any:
 
         value = expression.unescaped_value
 
-        if not g.auth_user.has_rights(Capabilities.manage_users) and not g.auth_user.login == value:
-            raise ObjectNotFoundException("Currently you can query only for your own uploads.")
+        uploader = (db.session.query(User)
+                              .join(User.groups)
+                              .filter(g.auth_user.is_member(Group.id))
+                              .filter(Group.name != "public")
+                              .filter(User.login == value)).scalar()
+        if g.auth_user.has_rights(Capabilities.manage_users):
+            uploader = db.session.query(User).filter(User.login == value).first()
```

Comment:
I would put the non-admin query under `else` block. Now we're doing two queries in admin case while the result of the first query is ignored. Also both queries should use the same method (`first` instead of `scalar`, my first suggestion was wrong)
```suggestion
        if g.auth_user.has_rights(Capabilities.manage_users):
            uploader = db.session.query(User).filter(User.login == value).first()
        else:
            uploader = (db.session.query(User)
                                  .join(User.groups)
                                  .filter(g.auth_user.is_member(Group.id))
                                  .filter(Group.name != "public")
                                  .filter(User.login == value)).first()
```